code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,899 @@
+			revert_cast()
+			return
+		if(is_shadow_or_thrall(target))
+			user << "<span class='warning'>You cannot glare at allies!</span>"
+			revert_cast()
+			return
+		var/mob/living/L = user
+		if(L.incorporeal_move) //Other abilities can still be used, but glare needed balancing
+			user << "<span class='warning'>You cannot glare while shadow walking!</span>"
+			revert_cast()
+			return
+		var/mob/living/carbon/human/M = target
+		user.visible_message("<span class='warning'><b>[user]'s eyes flash a blinding red!</b></span>")
+		target.visible_message("<span class='danger'>[target] freezes in place, their eyes glazing over...</span>")
+		if(in_range(target, user))
+			target << "<span class='userdanger'>Your gaze is forcibly drawn into [user]'s eyes, and you are mesmerized by the heavenly lights...</span>"
+		else //Only alludes to the shadowling if the target is close by
+			target << "<span class='userdanger'>Red lights suddenly dance in your vision, and you are mesmerized by their heavenly beauty...</span>"
+		target.Stun(10)
+		M.silent += 10
+
+
+/obj/effect/proc_holder/spell/aoe_turf/veil //Puts out most nearby lights except for flares and yellow slime cores
+	name = "Veil"
+	desc = "Extinguishes most nearby light sources."
+	panel = "Shadowling Abilities"
+	charge_max = 150 //Short cooldown because people can just turn the lights back on
+	human_req = 1
+	clothes_req = 0
+	range = 5
+	action_icon_state = "veil"
+	var/blacklisted_lights = list(/obj/item/device/flashlight/flare, /obj/item/device/flashlight/slime)
+	var/admin_override = 0 //Requested by Shadowlight213. Allows anyone to cast the spell, not just shadowlings.
+
+/obj/effect/proc_holder/spell/aoe_turf/veil/proc/extinguishItem(obj/item/I) //Does not darken items held by mobs due to mobs having separate luminosity, use extinguishMob() or write your own proc.
+	if(istype(I, /obj/item/device/flashlight))
+		var/obj/item/device/flashlight/F = I
+		if(F.on)
+			if(is_type_in_list(I, blacklisted_lights))
+				I.visible_message("<span class='danger'>[I] dims slightly before scattering the shadows around it.</span>")
+				return F.brightness_on //Necessary because flashlights become 0-luminosity when held.  I don't make the rules of lightcode.
+			F.on = 0
+			F.update_brightness()
+	else if(istype(I, /obj/item/device/pda))
+		var/obj/item/device/pda/P = I
+		P.fon = 0
+	I.set_light(0)
+	return I.luminosity
+
+/obj/effect/proc_holder/spell/aoe_turf/veil/proc/extinguishMob(mob/living/H)
+	var/blacklistLuminosity = 0
+	for(var/obj/item/F in H)
+		blacklistLuminosity += extinguishItem(F)
+	H.set_light(blacklistLuminosity) //I hate lightcode for making me do it this way
+
+/obj/effect/proc_holder/spell/aoe_turf/veil/cast(list/targets,mob/user = usr)
+	if(!shadowling_check(user) && !admin_override)
+		revert_cast()
+		return
+	user << "<span class='shadowling'>You silently disable all nearby lights.</span>"
+	for(var/turf/T in targets)
```

Comment:
I don't know, there's a lot of objects that produce light. Most don't have a proc for it, and you'll need to make one.